vlingo-reactive-messaging-patterns/src/main/java/io/vlingo/reactive/messaging/patterns/messagebus/TradingBus.java

```diff
@@ -0,0 +1,176 @@
+// Copyright © 2012-2018 Vaughn Vernon. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the
+// Mozilla Public License, v. 2.0. If a copy of the MPL
+// was not distributed with this file, You can obtain
+// one at https://mozilla.org/MPL/2.0/.
+package io.vlingo.reactive.messaging.patterns.messagebus;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Vector;
+
+import io.vlingo.actors.testkit.TestUntil;
+import io.vlingo.reactive.messaging.patterns.messagebus.exception.InvalidCommandIdException;
+import io.vlingo.reactive.messaging.patterns.messagebus.exception.InvalidNotificationIdException;
+
+/**
+ * TradingBus is the nexus between the {@link StockTrader}, {@link PortfolioManager}, and 
+ * {@link MarketAnalysisTools} systems providing an ability to register interest in specific types
+ * of messages and then delivering these messages asynchronously to each registrant.
+ *
+ * @author brsg.io
+ * @since Oct 31, 2018
+ */
+public class TradingBus 
+extends AbstractTradingActor
+//implements TradingBusProcessor
+{
+    private final Map<String, Vector<CommandHandler>> commandHandlers = new HashMap<>();
+    private final Map<String, Vector<NotificationInterest>> notificationInterests = new HashMap<>();
+    
+    public TradingBus( TestUntil until )
+    {
+        super( until );
+    }
+    
+    /* @see io.vlingo.reactive.messaging.patterns.messagebus.AbstractTradingActor#initialize() */
+    @Override
+    void initialize()
+    {
+        logger().log( "TradingBus initialize nothing to do" );
+    }
+    
+    /* @see io.vlingo.reactive.messaging.patterns.messagebus.AbstractTradingActor#executeBuyOrder(java.lang.String, java.lang.String, java.lang.Integer, java.lang.Double) */
+    @Override
+    public void executeBuyOrder( String portfolioId, String symbol, Integer quantity, Double price )
+    {
+        logger().log( String.format( "%s::executeBuyOrder( %s, %s, %d, %.2f )", getClass().getSimpleName(), portfolioId, symbol, quantity, price ));
+        Vector<CommandHandler> commandHandlerVector = commandHandlers.get( TradingProcessor.EXECUTE_BUY_ORDER );
+        for ( CommandHandler commandHandler : commandHandlerVector )
+        {
+            TradingProcessor tp = commandHandler.getTradingProcessor();
+            tp.executeBuyOrder( portfolioId, symbol, quantity, price );
+        }
+        
+        getUntil().happened();
```

Comment:
For example, `getUntil().happended()` is a computer instruction,  `until().happended()` is an expression.